ext/intl/config.m4

```diff
@@ -80,7 +80,28 @@ if test "$PHP_INTL" != "no"; then
     breakiterator/codepointiterator_methods.cpp"
 
   PHP_REQUIRE_CXX()
-  PHP_CXX_COMPILE_STDCXX(11, mandatory, PHP_INTL_STDCXX)
+
+  AC_MSG_CHECKING([if intl requires -std=gnu++17])
+  AC_RUN_IFELSE([AC_LANG_SOURCE([[
+#include <unicode/uvernum.h>
+
+int main(int argc, char *argv[])
+{
+#if U_ICU_VERSION_MAJOR_NUM >= 74
+    return 0;
+#else
+    return 1;
+#endif
+}
+  ]])],[
+    AC_MSG_RESULT([yes])
+    PHP_CXX_COMPILE_STDCXX(17, mandatory, PHP_INTL_STDCXX)
+  ],[
+    AC_MSG_RESULT([no])
+    PHP_CXX_COMPILE_STDCXX(11, mandatory, PHP_INTL_STDCXX)
+  ], [
+  ])
```

Comment:
```suggestion
  AS_IF([test "$PKG_CONFIG icu-uc --atleast-version=74"],[
    AC_MSG_RESULT([yes])
    PHP_CXX_COMPILE_STDCXX(17, mandatory, PHP_INTL_STDCXX)
  ],[
    AC_MSG_RESULT([no])
    PHP_CXX_COMPILE_STDCXX(11, mandatory, PHP_INTL_STDCXX)
  ])
```

Perhaps instead of running the test program, this can be checked directly from the package.

Otherwise, also something like this can be done:

```m4
AC_COMPILE_IFELSE([AC_LANG_PROGRAM([#include <unicode/uvernum.h>],[[
  #if U_ICU_VERSION_MAJOR_NUM >= 74
    return 0;
  #else
  # error "ICU version < 74"
  #endif
]])],[
  AC_MSG_RESULT([yes])
  PHP_CXX_COMPILE_STDCXX(17, mandatory, PHP_INTL_STDCXX)
],[
  AC_MSG_RESULT([no])
  PHP_CXX_COMPILE_STDCXX(11, mandatory, PHP_INTL_STDCXX)
])
```